packages/react/src/hooks/useDataState.ts

```diff
@@ -0,0 +1,42 @@
+import React from "react";
+
+interface ActionState<T> {
+  data: T;
+  isLoading: boolean;
+  message: string | undefined;
+}
+
+const getActionState = <T>(data: T): ActionState<T> => ({
+  data,
+  isLoading: false,
+  message: undefined,
+});
+
+export function useDataState<T, K>(
```

Comment:
`useDataState` is available in `@aws-amplify/ui-react-core`, no need to duplicate it